You have test failures:
```sh
       > FAILED tests/test_pregel.py::test_repeat_condition - AssertionError: assert [+ received] == [- snapshot]
       > FAILED tests/test_pregel.py::test_nested_graph_xray - AssertionError: assert [+ received] == [- snapshot]
```

These look like implementation tests instead of true unit tests, so I would suggest disabling them. Possibly also file a bug on the source repos.

I also suggest merging #940036 into this and adding `langchain-core` to the update script. I'll go look at the langchain and langsmith release logs and see if we should add additional packages to the update script.

--- a/pkgs/development/python-modules/langchain-core/default.nix
+++ b/pkgs/development/python-modules/langchain-core/default.nix
@@ -3,38 +3,45 @@
   stdenv,
   buildPythonPackage,
   fetchFromGitHub,
+
+  # build-system
+  poetry-core,
+
+  # dependencies
+  jsonpatch,
+  langsmith,
+  packaging,
+  pyyaml,
+  tenacity,
+
+  # optional-dependencies
+  pydantic,
+
+  # tests
   freezegun,
   grandalf,
   httpx,
-  jsonpatch,
-  langsmith,
   numpy,
-  packaging,
-  poetry-core,
-  pydantic,
   pytest-asyncio,
   pytest-mock,
   pytest-xdist,
   pytestCheckHook,
-  pythonOlder,
-  pyyaml,
   syrupy,
-  tenacity,
+
+  # passthru
   writeScript,
 }:
 
 buildPythonPackage rec {
   pname = "langchain-core";
-  version = "0.2.37";
+  version = "0.2.38";
   pyproject = true;
-
-  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langchain";
     rev = "refs/tags/langchain-core==${version}";
-    hash = "sha256-An2ApN0pgCrQjqu9XPFfPyPvWx0+6JnUkGPrcD0/3kg=";
+    hash = "sha256-3nRirzQe5KCVoeJ29fYelYuOD6r4adJof4NXreyfrzY=";
   };
 
   sourceRoot = "${src.name}/libs/core";

--- a/pkgs/development/python-modules/langchain-text-splitters/default.nix
+++ b/pkgs/development/python-modules/langchain-text-splitters/default.nix
@@ -2,25 +2,28 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+
+  # build-system
   poetry-core,
+
+  # dependencies
   langchain-core,
+
+  # tests
   pytest-asyncio,
   pytestCheckHook,
-  pythonOlder,
 }:
 
 buildPythonPackage rec {
   pname = "langchain-text-splitters";
-  version = "0.2.2";
+  version = "0.2.4";
   pyproject = true;
-
-  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langchain";
     rev = "refs/tags/langchain-text-splitters==${version}";
-    hash = "sha256-SixF3ZkN+gjQ4KYLhGoezdQAOQ1AlGEC6IBzHePF6/o=";
+    hash = "sha256-8n5eImRXOG/3tN/59Gd2/GpoGpt7P2ABj0T4pJi6xrk=";
   };
 
   sourceRoot = "${src.name}/libs/text-splitters";

--- a/pkgs/development/python-modules/langchain/default.nix
+++ b/pkgs/development/python-modules/langchain/default.nix
@@ -1,46 +1,54 @@
 {
   lib,
-  aiohttp,
-  async-timeout,
-  bash,
   buildPythonPackage,
   fetchFromGitHub,
-  freezegun,
+  pythonOlder,
+
+  # build-system
+  poetry-core,
+
+  # buildInputs
+  bash,
+
+  # dependencies
+  aiohttp,
   langchain-core,
   langchain-text-splitters,
   langsmith,
-  lark,
-  numpy,
-  pandas,
-  poetry-core,
   pydantic,
+  pyyaml,
+  requests,
+  sqlalchemy,
+  tenacity,
+  async-timeout,
+
+  # optional-dependencies
+  numpy,
+
+  # tests
+  freezegun,
+  lark,
+  pandas,
   pytest-asyncio,
   pytest-mock,
   pytest-socket,
   pytestCheckHook,
-  pythonOlder,
-  pyyaml,
   requests-mock,
-  requests,
   responses,
-  sqlalchemy,
   syrupy,
-  tenacity,
   toml,
 }:
 
 buildPythonPackage rec {
   pname = "langchain";
-  version = "0.2.15";
+  version = "0.2.16";
   pyproject = true;
-
-  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langchain";
     rev = "refs/tags/langchain==${version}";
-    hash = "sha256-8F6ntFstCTQjQNbE9oiYbpZ7kZ1grcnV3FHAfhFnAzA=";
+    hash = "sha256-8n5eImRXOG/3tN/59Gd2/GpoGpt7P2ABj0T4pJi6xrk=";
   };
 
   sourceRoot = "${src.name}/libs/langchain";

